docs/architecture/adr-versions.md

```diff
@@ -0,0 +1,219 @@
+# Protocol and Chain Versions
+
+The Software Version is covered by SemVer and described elsewhere.
+It is not relevant to the protocol description, suffice to say that if any protocol version
+changes, the software version changes. Software version is included in NodeInfo for convenience/diagnostics.
```

Comment:
Now update version must change code： https://github.com/tendermint/tendermint/commit/2fa99628351bd6d4ba14226bb7336135d655ff40.


There is a problem, if  already run old version(major  version different) nodes，and many user use them everyday. I can't upgrade one node which use new version code. 

In my opinion , new version should compatible old version. At least it should not be due to the issue of version number, such as Ethereum 、 Bitcoin

Reply:
Once we introduce protocol versions, the software version won't be relevant for peer connectivity - just the protocol versions.